packages/contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -83,12 +89,28 @@ contract OptimismPortal is ResourceMetering {
     mapping(bytes32 => bool) public finalizedWithdrawals;
 
     /**
-     * @param _l2Oracle                  Address of the L2OutputOracle.
-     * @param _finalizationPeriodSeconds Finalization time in seconds.
+     * @notice Reserve extra slots (to to a total of 50) in the storage layout for future upgrades.
+     */
+    uint256[48] private __gap;
+
+    /**
+     * @notice The constructor sets immutable values in the implementation.
+     *         This means that these values can only changed by an upgrade. But the efficiency gains
+     *         are worthwhile.
+     *         Also ensures that the implementation is initialized upon deployment.
      */
     constructor(L2OutputOracle _l2Oracle, uint256 _finalizationPeriodSeconds) {
+        // Set these immutable values into the bytcode of the implementation.
         L2_ORACLE = _l2Oracle;
         FINALIZATION_PERIOD_SECONDS = _finalizationPeriodSeconds;
+        initialize();
```

Comment:
OZ recommends this pattern in the constructor. It would break any existing testing infra, we would need to write the code that sets the proxy in both the solidity tests + the go tests.

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/83277ff916ac4f58fec072b8f28a252c1245c2f1/contracts/proxy/utils/Initializable.sol#L52

Reply:
I'm not super opinionated, curious what you think

Reply:
If they *must* be deployed behind proxies, then we should test them behind proxies in the unit tests. All of the proxy stuff could be done in the common tests. We will also need to update the hh deploy scripts to deploy proxies